File: account/java/crypt.java
import javax.crypto.Cipher;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;
import java.nio.charset.StandardCharsets;
import java.util.Base64;

public class Main {

    private static final int BLOCK_SIZE = 32;

    public static byte[] pKCS7Padding(String plaintext, int blockSize) {
        int padding = blockSize - (plaintext.length() % blockSize);
        byte padValue = (byte) padding;
        byte[] padtext = new byte[padding];
        for (int i = 0; i < padding; i++) {
            padtext[i] = padValue;
        }
        return (plaintext + new String(padtext, StandardCharsets.UTF_8)).getBytes(StandardCharsets.UTF_8);
    }

    public static byte[] pKCS7Unpadding(byte[] plaintext, int blockSize) {
        int padding = plaintext[plaintext.length - 1];
        if (padding > blockSize || padding > plaintext.length) {
            throw new IllegalArgumentException("Invalid padding");
        }
        return java.util.Arrays.copyOf(plaintext, plaintext.length - padding);
    }

    public static byte[] CbcEncrypt(String plaintext, String encodingAes) throws Exception {
        byte[] aeskey = Base64.getDecoder().decode(encodingAes + "=");
        int blockSize = 32;
        byte[] padMsg = pKCS7Padding(plaintext, blockSize);

        SecretKeySpec secretKeySpec = new SecretKeySpec(aeskey, "AES");
        byte[] iv = java.util.Arrays.copyOf(aeskey, 16);
        IvParameterSpec ivParameterSpec = new IvParameterSpec(iv);

        Cipher cipher = Cipher.getInstance("AES/CBC/NoPadding");
        cipher.init(Cipher.ENCRYPT_MODE, secretKeySpec, ivParameterSpec);

        byte[] ciphertext = cipher.doFinal(padMsg);
        return Base64.getEncoder().encode(ciphertext);
    }

    public static byte[] CbcDecrypt(String base64EncryptMsg, String encodingAes) throws Exception {
        byte[] aeskey = Base64.getDecoder().decode(encodingAes + "=");
        byte[] encryptMsg = Base64.getDecoder().decode(base64EncryptMsg);
        int blockSize = 32;

        SecretKeySpec secretKeySpec = new SecretKeySpec(aeskey, "AES");
        byte[] iv = java.util.Arrays.copyOf(aeskey, 16);
        IvParameterSpec ivParameterSpec = new IvParameterSpec(iv);

        Cipher cipher = Cipher.getInstance("AES/CBC/NoPadding");
        cipher.init(Cipher.DECRYPT_MODE, secretKeySpec, ivParameterSpec);

        byte[] decrypted = cipher.doFinal(encryptMsg);
        return pKCS7Unpadding(decrypted, blockSize);
    }

    public static void main(String[] args) {
        String plaintext = "1111";
        String encodingAes = "0VShOOAlYzVbKvcU0JRfsEdArPPunB5B6CMoFdAYVXa";

        try {
            byte[] secretText = CbcEncrypt(plaintext, encodingAes);
            System.out.println("密文：" + new String(secretText, StandardCharsets.UTF_8));

            byte[] text = CbcDecrypt(new String(secretText, StandardCharsets.UTF_8), encodingAes);
            System.out.println("明文：" + new String(text, StandardCharsets.UTF_8));
        } catch (Exception e) {
            System.out.println("加解密失败：" + e.getMessage());
        }
    }
}

File: parameter/go/crypt.go
package main

import (
	"bytes"
	"crypto/aes"
	"crypto/cipher"
	"encoding/base64"
	"errors"
	"fmt"
)

// pKCS7Padding 加密
func pKCS7Padding(plaintext string, blockSize int) []byte {
	padding := blockSize - (len(plaintext) % blockSize)
	padtext := bytes.Repeat([]byte{byte(padding)}, padding)
	var buffer bytes.Buffer
	buffer.WriteString(plaintext)
	buffer.Write(padtext)
	return buffer.Bytes()
}

// pKCS7Padding 解密
func pKCS7Unpadding(plaintext []byte, block_size int) ([]byte, error) {
	plaintext_len := len(plaintext)
	if nil == plaintext || plaintext_len == 0 {
		return nil, errors.New("pKCS7Unpadding error nil or zero")
	}
	if plaintext_len%block_size != 0 {
		return nil, errors.New("pKCS7Unpadding text not a multiple of the block size")
	}
	padding_len := int(plaintext[plaintext_len-1])
	if len(plaintext) < plaintext_len-padding_len || plaintext_len-padding_len < 0 {
		return nil, errors.New("pKCS7Unpadding plaintext len too small")
	}
	return plaintext[:plaintext_len-padding_len], nil
}

// CbcEncrypt 加密
func CbcEncrypt(plaintext, encodingAes string) ([]byte, error) {
	aeskey, err := base64.StdEncoding.DecodeString(encodingAes + "=")
	if nil != err {
		return nil, err
	}
	const blockSize = 32
	padMsg := pKCS7Padding(plaintext, blockSize)

	block, err := aes.NewCipher(aeskey)
	if err != nil {
		return nil, err
	}

	ciphertext := make([]byte, len(padMsg))
	iv := aeskey[:aes.BlockSize]

	mode := cipher.NewCBCEncrypter(block, iv)

	mode.CryptBlocks(ciphertext, padMsg)
	base64Msg := make([]byte, base64.StdEncoding.EncodedLen(len(ciphertext)))
	base64.StdEncoding.Encode(base64Msg, ciphertext)

	return base64Msg, nil
}

// CbcDecrypt 解密
func CbcDecrypt(base64EncryptMsg, encodingAes string) ([]byte, error) {
	aeskey, err := base64.StdEncoding.DecodeString(encodingAes + "=")
	if nil != err {
		return nil, err
	}
	encrypt_msg, err := base64.StdEncoding.DecodeString(base64EncryptMsg)
	if err != nil {
		return nil, err
	}

	block, err := aes.NewCipher(aeskey)
	if err != nil {
		return nil, err
	}

	if len(encrypt_msg) < aes.BlockSize {
		return nil, err
	}
	iv := aeskey[:aes.BlockSize]
	if len(encrypt_msg)%aes.BlockSize != 0 {
		return nil, err
	}
	mode := cipher.NewCBCDecrypter(block, iv)
	mode.CryptBlocks(encrypt_msg, encrypt_msg)

	const blockSize = 32
	encrypt_msg, err = pKCS7Unpadding(encrypt_msg, blockSize)
	return encrypt_msg, err
}

// CbcEncryptReturnStr 加密
func CbcEncryptReturnStr(plaintext, encodingAes string) (string, error) {
	bs, err := CbcEncrypt(plaintext, encodingAes)
	if err != nil {
		return plaintext, err
	}
	return string(bs), err
}

// CbcDecryptReturnStr 解密
func CbcDecryptReturnStr(base64EncryptMsg, encodingAes string) (string, error) {
	bs, err := CbcDecrypt(base64EncryptMsg, encodingAes)
	if err != nil {
		return base64EncryptMsg, err
	}
	return string(bs), err
}

func main() {
	plaintext := "1111"
	aesKey := "0VShOOAlYzVbKvcU0JRfsEdArPPunB5B6CMoFdAYVXa"

	encryptMsg, err := CbcEncryptReturnStr(plaintext, aesKey)
	if err != nil {
		fmt.Println("err:", err.Error())
		return
	}
	fmt.Println("encryptMsg:", encryptMsg)

	decryptMsg, err := CbcDecryptReturnStr(encryptMsg, aesKey)
	if err != nil {
		fmt.Println("err:", err.Error())
		return
	}
	fmt.Println("decryptMsg:", decryptMsg)
}
